pkg/buildpacks/telemetry.go

```diff
@@ -0,0 +1,155 @@
+package buildpacks
+
+import (
+	"encoding/json"
+
+	"github.com/SAP/jenkins-library/pkg/cnbutils/privacy"
+	"github.com/SAP/jenkins-library/pkg/cnbutils/project"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+	"github.com/pkg/errors"
+)
+
+const version = 3
+
+type Telemetry struct {
+	customData *telemetry.CustomData
+	data       *BuildpacksTelemetry
+}
+
+func NewTelemetry(customData *telemetry.CustomData) Telemetry {
+	return Telemetry{
+		customData: customData,
+		data: &BuildpacksTelemetry{
+			Version: version,
+		},
+	}
+}
+
+func (d *Telemetry) Export() error {
+	d.customData.Custom1Label = "cnbBuildStepData"
+	customData, err := json.Marshal(d.data)
+	if err != nil {
+		return errors.Wrap(err, "failed to marshal custom telemetry data")
+	}
+	d.customData.Custom1 = string(customData)
+	return nil
+}
+
+func (d *Telemetry) WithImage(image string) {
```

Comment:
Should we make it a fluent API by just also returning `*Telemetry`?

I kind of always expect this from `WithX()`. But we wouldn't make much use of it here.